kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,302 @@
+                    _logger.exception(f'Import failed: this parameter cannot be modified on this device: {reg_check}')
+                    return CheckResult.NOT_MUTABLE, reg_check.name
+
+                elif not check_type(reg_check, imported_registers[reg_check.name]):
+                    _logger.exception(f'Import failed: this parameter type is incorrect {reg_check}')
+                    return CheckResult.INCORRECT_TYPE, reg_check.name
+
+                elif len(imported_registers[reg_check.name]) != len(reg_check.cached_value):
+                    _logger.exception(f'Import failed: this parameter dimension is incorrect {reg_check}')
+                    return CheckResult.INCORRECT_DIMENSION, reg_check.name
+
+                elif reg_check.has_min_and_max_values and reg_check.type_id != Register.ValueType.BOOLEAN:
+                    if not (reg_check.min_value <= imported_registers[reg_check.name] <= reg_check.max_value):
+                        _logger.exception('Import failed: this parameter value is outside '
+                                          f'permitted range: {reg_check}')
+                        return CheckResult.OUTSIDE_RANGE, reg_check.name
+
+    except Exception as ex:
+        _logger.exception(f'Could not write registers: {str(ex)}')
+        return CheckResult.UNKNOWN, str(ex)
+
+    return CheckResult.NO_ERROR, ''
+
+
+def show_error_box(result, detail, file_name, parent):
+    show_error('Import failed',
+               f'Cannot import {file_name}',
+               CHECK_RESULT_MAPPING[result] + detail,
+               parent)
+
+
+def write_registers(parent, file_name, registers, imported_registers):
+    async def executor():
+        unwritten_registers = []
+        for reg in imported_registers:
+            _attempt = 0
+            while True:
+                try:
+                    old_reg = next((r for r in registers if r.name == reg))
+                    await old_reg.write_through(imported_registers[reg])
+                    break
+
+                except Exception as ex:
+                    _attempt += 1
+                    _logger.exception(f'Register {reg} could not be loaded (attempt {_attempt}/3)')
+                    if _attempt >= 3:
+                        try:
+                            old_reg = next((r for r in registers if r.name == reg))
+                            old_value = old_reg.cached_value
+                        except Exception:
+                            old_value = ['(unknown)']
+
+                        unwritten_registers.append([reg, old_value, imported_registers[reg]])
+                        break
+
+        if unwritten_registers:
+            display_warning_message(parent, unwritten_registers)
+
+        display_sucess_message('Import successful',
+                               f'{file_name} have been successfully imported',
+                               '',
+                               parent)
```

Comment:
We can't use blocking because that interferes with the event loop scheduling. To fix the problem you described, either remove the warning box completely and add its data into the success box, or spawn a new thread and display the boxes sequentially from there (not a great idea, too complex with little benefit).